xmrstak/net/jpsock.cpp

```diff
@@ -94,7 +94,9 @@ struct jpsock::opq_json_val
 	opq_json_val(const Value* val) : val(val) {}
 };
 
-jpsock::jpsock(size_t id, bool tls) : pool_id(id)
+jpsock::jpsock(size_t id, const char* sAddr, const char* sLogin, const char* sPassword, double pool_weight, bool dev_pool, bool tls, const char* tls_fp, bool nicehash) :
```

Comment:
could you please use `std::string` this is more save and this part is not time critical.

Reply:
Since the JSON lib returns const char* I prefer to keep it this way until we need to store the string (when the constructor runs). I found that many people find c++ rvalue refs confusing.